From c8f01435114f8bd4211b8ff92f3fa96f0e6b55d4 Mon Sep 17 00:00:00 2001
From: Stephan Gerhold <stephan@gerhold.net>
Date: Wed, 22 Sep 2021 21:58:53 +0200
Subject: [PATCH 011/210] UPSTREAM: arm64: dts: qcom: apq8016-sbc: Clarify
 firmware-names

Commit 0f6b380d580c ("arm64: dts: qcom: apq8016-sbc: Update modem and WiFi
firmware path") added "firmware-name"s to the APQ8016 SBC (DB410c) device
tree to separate the (test key)-signed firmware from other devices.

However, the added names are a bit confusing. The "modem" firmware used by
DB410c is actually a simplified version for APQ8016 that lacks most of the
modem functionality (phone calls, SMS etc) that is available on MSM8916.
Placing it in "qcom/msm8916/modem.mbn" suggests that it supports all
functionality for MSM and not just the reduced functionality for APQ.

Request the firmware from "qcom/apq8016/modem.mbn" instead to clarify this.
Do the same for "wcnss.mbn" for consistency (although the WCNSS firmware
works just fine on MSM8916).

Finally, add a "_sbc" suffix to the WCNSS_qcom_wlan_nv.bin firmware file.
It seems like the nv.bin firmware is somewhat board specific and can
therefore vary a bit from device to device. This makes it more clear
which board it is intended to be used for.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
Signed-off-by: Bjorn Andersson <bjorn.andersson@linaro.org>
Link: https://lore.kernel.org/r/20210922195853.95574-1-stephan@gerhold.net
(cherry picked from commit 2533786f46d074d67a4bca04c2d44d3825594415)
---
 arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
index 351c68d29af..0e4a1f00402 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
+++ b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
@@ -308,7 +308,7 @@ &mdss {
 &mpss {
 	status = "okay";
 
-	firmware-name = "qcom/msm8916/mba.mbn", "qcom/msm8916/modem.mbn";
+	firmware-name = "qcom/apq8016/mba.mbn", "qcom/apq8016/modem.mbn";
 };
 
 &pm8916_resin {
@@ -319,7 +319,7 @@ &pm8916_resin {
 &pronto {
 	status = "okay";
 
-	firmware-name = "qcom/msm8916/wcnss.mbn";
+	firmware-name = "qcom/apq8016/wcnss.mbn";
 };
 
 &sdhc_1 {
@@ -403,7 +403,7 @@ &wcd_codec {
 };
 
 &wcnss_ctrl {
-	firmware-name = "qcom/msm8916/WCNSS_qcom_wlan_nv.bin";
+	firmware-name = "qcom/apq8016/WCNSS_qcom_wlan_nv_sbc.bin";
 };
 
 /* Enable CoreSight */
-- 
2.34.1

